### PR TITLE
Notebook fixes

### PR DESCRIPTION
**Pull request by jklaise**
Fixes most errors found in #330 and #323.

### CODE REVIEW

**Comment by jklaise** [comment]
The last commit fixes `cd_ks_cifar10.ipynb` by chaning the import path from `alibi_detect.models.resnet` `alibi_detect.models.tensorflow.resnet`. I note that an equivalent function does not exist for `pytorch` (as a consequence of no implementation of `resnet`) so maybe we need to clarify what it means when the notebook says that `pytorch` is supported? @arnaudvl

File: examples/cd_distillation_cifar10.ipynb [hunk 1]
```diff
@@ -30,17 +30,9 @@
   },
```
Comment (ascillitoe) [inline examples/cd_distillation_cifar10.ipynb:30]: <pre class="ql-syntax" spellcheck="false">Line #9.    y_pred = predict_batch(clf, X_test, batch_size=256, return_class=True)
</pre><p> </p><p>Returns: </p><pre class="ql-syntax" spellcheck="false">TypeError: predict_batch() got an unexpected keyword argument 'return_class'
</pre><p> </p><p>See first bullet point in #323 (predict_batch args were changed when made tensorflow/pytorch specific).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi-detect/pull/333/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='26'/>

File: examples/cd_spot_the_diff_mnist_wine.ipynb [hunk 2]
```diff
@@ -48,16 +48,27 @@
    "cell_type": "code",
```
Comment (ascillitoe) [inline examples/cd_spot_the_diff_mnist_wine.ipynb:48]: <pre class="ql-syntax" spellcheck="false">Line #7.    mnist_train_ds = torchvision.datasets.MNIST(MNIST_PATH, train=True, download=DOWNLOAD)
</pre><p> </p><p>Shall we strip out these <code>UserWarning</code>  outputs for the docs?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi-detect/pull/333/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (ascillitoe) [inline examples/cd_spot_the_diff_mnist_wine.ipynb:48]: <pre class="ql-syntax" spellcheck="false">Line #8.    all_x, all_y = mnist_train_ds.train_data, mnist_train_ds.target
</pre><p> </p><p>GIves:</p><pre class="ql-syntax" spellcheck="false">`AttributeError: 'MNIST' object has no attribute 'target'
</pre><p> </p><p>Possibly  a <code>torchvision</code>  versioning issue (I'm on <code>0.10.0+cpu</code>).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SeldonIO/alibi-detect/pull/333/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

**Review by ascillitoe — COMMENTED** [review]
These changes generally look good. Only newly introduced issue that might be good to address before merging is the user-specific output/warning in `cd_spot_the_diff_mnist_wine.ipynb`?

Also, just to add another issue to the mix, there's possibly a torchvision versioning issue in `cd_spot_the_diff_mnist_wine.ipynb` (will add to list in #323).

**Comment by jklaise** [comment]
Thanks, I'll look over the warnings once more.

Strange about the `torchvision` one as `train_labels` was deprecated in favour of `target` ~3 years ago, will double check with 0.10.

File: examples/cd_spot_the_diff_mnist_wine.ipynb (hunk 2 above)
Comment (ascillitoe) [reply examples/cd_spot_the_diff_mnist_wine.ipynb:48]: Correction: Above is nothing to do with `torchvision` versioning, just a typo. 

`all_x, all_y = mnist_train_ds.train_data, mnist_train_ds.target`

should be:

`all_x, all_y = mnist_train_ds.train_data, mnist_train_ds.targets`

Comment (jklaise) [reply examples/cd_spot_the_diff_mnist_wine.ipynb:48]: Ah, I did see that and thought I fixed it already.

File: examples/cd_distillation_cifar10.ipynb (hunk 1 above)
Comment (jklaise) [reply examples/cd_distillation_cifar10.ipynb:30]: Somehow missed that, this is why we need notebook tests!

**Review by ascillitoe — APPROVED** [review]

**Pull request merged**